test/time_evolvers/test_trotter_qrte.py

```diff
@@ -211,6 +211,30 @@ def test_trotter_qrte_trotter_hamiltonian_errors(self, operator, initial_state):
         """Test TrotterQRTE with raising errors for evolution problem content."""
         self._run_error_test(initial_state, operator, None, None, None, None)
 
+    def test_barriers(self):
```

Comment:
Could you parameterize this test using @ddt - you can see other tests here as examples if needed - so that passing in False and True as data values it can test both aspects so we can make sure when no barriers are asked for that indeed does happen too.